feat!: add `@vitest/coverage-v8` package (#3339)
vitest-dev/vitest: docs/config/index.md

### coverage<NonProjectOption />

> You can use [`c8`](https://github.com/bcoe/c8), [`v8`](https://v8.dev/blog/javascript-code-coverage), [`istanbul`](https://istanbul.js.org/)  or [a custom coverage solution](/guide/coverage#custom-coverage-provider) for coverage collection.

Reviewer: ```suggestion
You can use [`v8`](https://v8.dev/blog/javascript-code-coverage), [`istanbul`](https://istanbul.js.org/)  or [a custom coverage solution](/guide/coverage#custom-coverage-provider) for coverage collection.
```

We could start not listing it.

> ::: info
> The `c8` provider is being replaced by `v8` and will eventually be removed completely.
> :::

Reviewer: ```suggestion
::: info
The `c8` provider is being replaced by the `v8` provider. It will be deprecated in the next major version.
:::
```

#### coverage.provider

> - **Type:** `'c8' | 'v8' | 'istanbul' | 'custom'`
> - **Default:** `'c8'`

Reviewer: ```suggestion
- **Type:** `'v8' | 'istanbul' | 'custom'`
- **Default:** `'v8'`
```

Do we see any blocker we could make `v8` the new default without breaking existing usage?

Reviewer: Sounds good to me!

Author: There is now a message printed before prompting the installation of new package:

<img width="480px" src="https://github.com/vitest-dev/vitest/assets/14806298/d1401144-a684-4b86-aaa1-8d25c46a9f4c" />

This is only shown when `test.coverage.provider` or `--coverage.provider` option is not explicitly set.
